### PR TITLE
jing-dong-code: add new extension

### PR DESCRIPTION
**Pull request by zooPanda**
## Description
For generation or parsing the JD code

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zooPanda 👋 

Thanks for your contribution 💪 

At the moment, Raycast doesn't support localization and only supports US English – could you please correct this, so everything inside the extension is US English

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by zooPanda** [comment]
> Hi @zooPanda 👋
> 
> Thanks for your contribution 💪
> 
> At the moment, Raycast doesn't support localization and only supports US English – could you please correct this, so everything inside the extension is US English
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

it  was parsed into US English .

File: extensions/jing-kou-ling/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "title": "JD code",
+  "description": "generate and parse JD code",
+  "icon": "jdCodeIcon.png",
+  "author": "zooPanda",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "enjcode",
+      "title": "generate JD code",
+      "subtitle": "generate JD code",
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/package.json:13]: Use subtitles to add context to your command. Usually, it's an app or service name that you integrate with. It makes command names more lightweight and removes the need to specify a service name in the command title.

File: extensions/jing-kou-ling/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      "description": "generate JD code",
+      "mode": "view"
+    },
+    {
+      "name": "dejcode",
+      "title": "parse JD code",
+      "subtitle": "parse JD code",
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/package.json:20]: Use subtitles to add context to your command. Usually, it's an app or service name that you integrate with. It makes command names more lightweight and removes the need to specify a service name in the command title.

File: extensions/jing-kou-ling/src/dejcode.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+function DecodeItem({ title, value }: { title: string; value: string | number | undefined }) {
+  return (
+    <List.Item
+      title={title}
+      subtitle={`${value}`}
+      actions={
+        <ActionPanel>
+          <Action.CopyToClipboard title="复制到剪贴板" content={`${value}`} />
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/src/dejcode.tsx:36]: Needs to be translated to US English too

File: extensions/jing-kou-ling/src/enjcode.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      showToast({ title: "generate failed", message: data.msg, style: Toast.Style.Failure });
+    }
+  }
+
+  return (
+    <>
+      <Form
+        actions={
+          <ActionPanel>
+            <Action.SubmitForm onSubmit={handleSubmit} title="生成" />
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/src/enjcode.tsx:36]: Needs to be translated to US English too

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zooPanda 👋 

I have some feedback ready for you to address 💪 

- The panda is pretty hard to spot, could you enlarge the panda a bit, so it's possible to see? I think the logo works with both with and dark background, but let's see when you enhance it.

- Tried to encode something but only gets errors, do you have something I can try to input that works?

- Could you update(translations) and move screenshots to a folder in the root called `media`

- Consider adding a README 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/jing-kou-ling/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  "name": "jing-kou-ling",
+  "title": "JD code",
+  "description": "generate and parse JD code",
+  "icon": "jing-kou-ling-icon.png",
+  "author": "zooPanda",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "enjcode",
+      "title": "generate JD code",
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/package.json:12]:
```suggestion
      "title": "Generate JD Code",
```

File: extensions/jing-kou-ling/package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      "description": "use your custom title and url to generate JD code",
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/package.json:13]:
```suggestion
      "description": "Generate a JD code using your custom title and url",
```

File: extensions/jing-kou-ling/src/enjcode.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      <Form
+        actions={
+          <ActionPanel>
+            <Action.SubmitForm onSubmit={handleSubmit} title="generate" />
+          </ActionPanel>
+        }
+        isLoading={formLoading}
+      >
+        <Form.Description title="Generate JD code" text="Generate your own JD code" />
+        <Form.TextField id="title" title="title" placeholder="please enter your title " />
```
Comment (pernielsentikaer) [inline extensions/jing-kou-ling/src/enjcode.tsx:42]:
```suggestion
        <Form.TextField id="title" title="title" placeholder="Please enter your title " />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @zooPanda 👋 

Thanks for the updates 🔥 

I have some questions and suggestions for you 🐼 

1) What is a JD code, I simply don't understand it and I can't figure out what to search for
2) When generation of a JD succeed, you could consider popping to [root](https://developers.raycast.com/api-reference/window-and-search-bar#poptoroot)

Maybe a few more questions will follow when I fully understand what's happening 😄

**Comment by fe9lix** [comment]
Hey guys, I'm closing this because of inactivity. Feel free to reopen later and address the feedback.

**Pull request closed without merging**